src/config/target.rs

```diff
@@ -0,0 +1,19 @@
+use super::*;
+
+#[derive(Clone, Deserialize)]
+pub struct Target {
+    /// A list of target endpoints (`IP:PORT`).
+    endpoints: Vec<String>,
+    /// A cache name
+    cache_name: Option<String>,
+}
+
+impl Target {
+    pub fn endpoints(&self) -> &[String] {
+        &self.endpoints
+    }
+
+    pub fn cache_name(&self) -> Option<&str> {
```

Comment:
This looks a bit odd being part of a pretty general trait but clearly having special meaning.

Reply:
We may need to consider ways to provide this in the endpoints instead, as it maybe feels a little more natural to encode that type of information there. Redis has a provisional uri-scheme: https://www.iana.org/assignments/uri-schemes/prov/redis which can be used to include database id. Maybe we can adopt unofficial uri-schemes for other protocols as a way of providing this type of information?

Reply:
Oh that seems to be a really good idea! In the storage space it is widely used, so wouldn't be odd to most users.

Reply:
#11 opened to follow-up on this